common/src/main/java/org/jboss/pnc/common/util/UrlUtils.java

```diff
@@ -181,4 +181,39 @@ public static String stripProtocol(String url) {
         return (userInfo == null ? "" : userInfo + "@") + (host == null ? "" : host) + (port == -1 ? "" : ":" + port)
                 + (path == null ? "" : path) + queryAppend + fragmentAppend;
     }
+
+    /**
+     * Substitute the url with a new host (with new scheme and port), leaving the rest unchanged
+     *
+     * @param url Url to change host (e.g http://example.com/booya)
+     * @param newHost the new host (e.g https://localhost:1234)
+     *
+     * @return modified url
+     *
+     * @throws MalformedURLException If urls are malformed
+     */
+    public static String replaceHostInUrl(String url, String newHost) throws MalformedURLException {
+        if (url == null || newHost == null) {
+            return url;
+        }
+
+        URL originalUrl = new URL(url);
+        URL newHostUrl = new URL(newHost);
+
+        boolean hostHasPort = newHost.contains(":");
+
+        int newPort = -1;
+        if (hostHasPort) {
+            newPort = newHostUrl.getPort();
```

Comment:
@thescouser89 I believe the newHostUrl.getPort() would already return -1 if the port is not set, so probably LN203-208 can be replaced just with 
int newPort = newHostUrl.getPort();
?

Reply:
Thanks for catching this! I can confirm what you said haa. Good find! :)

PR: https://github.com/project-ncl/pnc/pull/3500